woocommerce/woocommerce-android
# Send `WCTracker` telemetry

Closes: #5196

Related FluxC: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2185

### Description

This PR is about sending basic data to `WCTracker`.

### Testing instructions

- [ ] TC1 Send WCTracker data
1. Assert that site you will launch app with is at WooCommerce >= 5.9.0
2. Open Flipper
2. Build and run app based on this branch
3. **Assert you can find WCTracker POST request** It's body looks like:
```
 body: "{"platform":"android","version":"7.9-rc-1"}"
 json: "true"
 path: "/wc-telemetry/tracker/&_method=post"
```
and response `data: null`

- [ ] TC2 Assert that time interval check works for same site
1. After TC1 restart app
2. **Assert there was *no* WCTracker request**

- [ ] TC3 Assert that time interval check works for different sites
1. After TC2 change site. It doesn't have to be WooCommerce >= 5.9.0
2. Go back to main screen
3. As we changed site, **assert you can find WCTracker POST request again**.
4. **Restart app - WCTracker POST request shouldn't be requested**.

- [ ] [Optional] TC4 If you have `ssh` access to website
E.g. on GoDaddy hosting, you can `ssh` into hosting and execute: `wp eval 'var_dump(WC_Tracker::get_tracking_data());'`. You should find a record like this:
```
["android"]=>
    array(3) {
      ["platform"]=>
      string(7) "android"
      ["version"]=>
      string(6) "7.9-rc-1"
      ["last_used"]=>
      string(25) "2021-11-09T22:41:34+00:00"
    }
```

### Images/gif

n/a
